server/integration/src/main/ant/infinispan-server.xml

```diff
@@ -53,127 +54,125 @@
+                        <equals arg1="${lsof.result}" arg2="0"/>
+                        <equals arg1="${jps.result}" arg2="0"/>
+                        <equals arg1="${cmd.result}" arg2="0"/>
                     </or>
                 </not>
             </condition>
         </fail>
     </target>
 
     <target name="-do.ps" if="${run.ps}">
-        <echo>this target will be called only when property $${run.ps} is set</echo>
-        <loadfile srcfile="ps.pid" property="ppid" failonerror="false">
-            <filterchain>
-                <prefixlines prefix=" " />
-                <striplinebreaks />
-                <trim />
-            </filterchain>
-        </loadfile>
         <exec executable="kill" osfamily="unix">
-            <arg line="-9 ${ppid}"/>
+            <arg line="-9 ${ps.pid}"/>
         </exec>
-        <delete file="ps.pid"/>
     </target>
 
     <target name="-do.lsof" if="${run.lsof}">
-        <echo>this target will be called only when property $${run.lsof} is set</echo>
         <exec executable="kill" osfamily="unix">
             <arg line="-9 ${lsof.pid}"/>
         </exec>
     </target>
 
     <target name="-do.jps" if="${run.jps}">
-        <echo>this target will be called only when property $${run.jps} is set</echo>
-        <loadfile srcfile="jps.pid" property="jpid" failonerror="false">
-            <filterchain>
-                <linecontains>
-                    <contains value="jboss-modules.jar"/>
-                </linecontains>
-                <tokenfilter>
-                    <deletecharacters chars="jboss-modules.jar"/>
-                    <ignoreblank/>
-                </tokenfilter>
-                <striplinebreaks/>
-            </filterchain>
-        </loadfile>
         <exec executable="kill" osfamily="unix">
-            <arg line="-9 ${jpid}"/>
+            <arg line="-9 ${jps.pid}"/>
         </exec>
-        <delete file="jps.pid"/>
     </target>
 
     <target name="-do.cmd" if="${run.cmd}">
-        <echo>this target will be called only when property $${run.cmd} is set</echo>
-        <loadfile srcfile="cmd.pid" property="cpid" failonerror="false">
-            <filterchain>
-                <linecontains>
-                    <contains value="LISTENING"/>
-                    <contains value=":${hotrod.port}"/>
-                </linecontains>
-                <tokenfilter>
-                    <replaceregex pattern=".*LISTENING([ \t]+)([0-9]+)" replace="\2" flags="gi"/>
-                    <ignoreblank/>
-                </tokenfilter>
-                <striplinebreaks/>
-            </filterchain>
-        </loadfile>
         <exec executable="taskkill" osfamily="windows">
-            <arg line="/F /T /PID ${cpid}"/>
+            <arg line="/F /T /PID ${cmd.pid}"/>
         </exec>
-        <delete file="cmd.pid"/>
     </target>
 
     <target name="kill-server" depends="-do.kinit, -do.ps, -do.lsof, -do.jps, -do.cmd"/>
```

Comment:
Suggestion: the server can be stopped on Linux with:
```
<HOME>/bin/ispn-cli.sh --connect command=:shutdown  
```

and on Windows:

```
<HOME>/bin/ispn-cli.bat --connect command=:shutdown
```

Reply:
You probably could get rid of all this logic coded in XML by simply calling the ispn-cli above, as it takes care of all the platform/JVM specific details  :)

Reply:
I'm not sure if it's still a problem nowadays but back when the scripts were first written we had problems with the server not stopping properly (https://issues.jboss.org/browse/WFLY-4433 I think). So even if we use the cli, we'd still have to keep `kill -9` as a backup.

Reply:
Provided that the ```kill -9``` would cause the test to fail rather than hide the problem, I'm fine with it.